### ISSUE TITLE
Recompilation fails even when nothing changed

### ISSUE THREAD

**Issue by ta1bbty** [iBotPeaches/Apktool#1991]
### Information
1. **Apktool Version (`apktool -version`)** - 2.3.4-dirty
2. **Operating System (Mac, Linux, Windows)** - Linux
3. **APK From? (Playstore, ROM, Other)** - Apkpure.com

### Stacktrace/Logcat

```
┌─[root@mypc]─[02:09:30 Thu Jan 17]─[/home/bbty/vivy]
└──> apktool d -f Vivy_v1.23.1_apkpure.com.apk
I: Using Apktool 2.3.4-dirty on Vivy_v1.23.1_apkpure.com.apk
I: Loading resource table...
I: Decoding AndroidManifest.xml with resources...
I: Loading resource table from file: /root/.local/share/apktool/framework/1.apk
I: Regular manifest package...
I: Decoding file-resources...
I: Decoding values */* XMLs...
I: Baksmaling classes.dex...
I: Baksmaling classes2.dex...
I: Baksmaling classes3.dex...
I: Baksmaling classes4.dex...
I: Baksmaling classes5.dex...
I: Baksmaling classes6.dex...
I: Copying assets and libs...
I: Copying unknown files...
I: Copying original files...
┌─[root@mypc]─[02:10:05 Thu Jan 17]─[/home/bbty/vivy]
└──> apktool b Vivy_v1.23.1_apkpure.com/ -o translated.apk
I: Using Apktool 2.3.4-dirty
I: Checking whether sources has changed...
I: Smaling smali folder into classes.dex...
Exception in thread "main" java.lang.NoSuchMethodError: java.nio.ByteBuffer.clear()Ljava/nio/ByteBuffer;
	at org.jf.dexlib2.writer.DexWriter.writeAnnotationDirectories(DexWriter.java:698)
	at org.jf.dexlib2.writer.DexWriter.writeTo(DexWriter.java:267)
	at org.jf.dexlib2.writer.DexWriter.writeTo(DexWriter.java:246)
	at brut.androlib.src.SmaliBuilder.build(SmaliBuilder.java:61)
	at brut.androlib.src.SmaliBuilder.build(SmaliBuilder.java:36)
	at brut.androlib.Androlib.buildSourcesSmali(Androlib.java:419)
	at brut.androlib.Androlib.buildSources(Androlib.java:350)
	at brut.androlib.Androlib.build(Androlib.java:302)
	at brut.androlib.Androlib.build(Androlib.java:269)
	at brut.apktool.Main.cmdBuild(Main.java:238)
	at brut.apktool.Main.main(Main.java:76)
```

### 2nd Stacktrace 
```
┌─[root@mypc]─[02:11:15 Thu Jan 17]─[/home/bbty/vivy]
└──> apktool d -r -f Vivy_v1.23.1_apkpure.com.apk
I: Using Apktool 2.3.4-dirty on Vivy_v1.23.1_apkpure.com.apk
I: Copying raw resources...
I: Baksmaling classes.dex...
I: Baksmaling classes2.dex...
I: Baksmaling classes3.dex...
I: Baksmaling classes4.dex...
I: Baksmaling classes5.dex...
I: Baksmaling classes6.dex...
I: Copying assets and libs...
I: Copying unknown files...
I: Copying original files...
┌─[root@mypc]─[02:13:49 Thu Jan 17]─[/home/bbty/vivy]
└──> apktool b -r Vivy_v1.23.1_apkpure.com/ -o translated.apk
I: Using Apktool 2.3.4-dirty
I: Checking whether sources has changed...
I: Smaling smali folder into classes.dex...
Exception in thread "main" java.lang.NoSuchMethodError: java.nio.ByteBuffer.clear()Ljava/nio/ByteBuffer;
	at org.jf.dexlib2.writer.DexWriter.writeAnnotationDirectories(DexWriter.java:698)
	at org.jf.dexlib2.writer.DexWriter.writeTo(DexWriter.java:267)
	at org.jf.dexlib2.writer.DexWriter.writeTo(DexWriter.java:246)
	at brut.androlib.src.SmaliBuilder.build(SmaliBuilder.java:61)
	at brut.androlib.src.SmaliBuilder.build(SmaliBuilder.java:36)
	at brut.androlib.Androlib.buildSourcesSmali(Androlib.java:419)
	at brut.androlib.Androlib.buildSources(Androlib.java:350)
	at brut.androlib.Androlib.build(Androlib.java:302)
	at brut.androlib.Androlib.build(Androlib.java:269)
	at brut.apktool.Main.cmdBuild(Main.java:238)
	at brut.apktool.Main.main(Main.java:76)
┌─[root@mypc]─[02:14:14 Thu Jan 17]─[/home/bbty/vivy]
└──> 
```
### Steps to Reproduce
1. `apktool `

### Frameworks
None

### APK
[Here](https://apkpure.com/vivy/eu.uvita)

### Questions to ask before submission
1. Have you tried `apktool d`, `apktool b` without changing anything? yes, that's what I'm reporting
2. If you are trying to install a modified apk, did you resign it? no
3. Are you using the latest apktool version? yes

### Screenshot 
When decompiling everythin  :
![image](https://user-images.githubusercontent.com/45117337/51277333-4ebde980-19fd-11e9-8587-439deafd80fe.png)

When skipping resources :

![image](https://user-images.githubusercontent.com/45117337/51277420-9fcddd80-19fd-11e9-90f0-7b0a59a49ee5.png)

**Comment by iBotPeaches** [comment]
Sorry for the delay. Since this crash is happening at the smali level (https://github.com/JesusFreke/smali), perhaps we fed invalid data to it or a bug upstream. I haven't had time to dig into it to know which.

**Comment by absane** [comment]
Hello. I am running into this issue as well. Due to an NDA I cannot release the APK, but I am getting the same error even if I don't change a single thing.

**Comment by iBotPeaches** [comment]
```
➜  1991 apktool d Vivy_v1.52.2_apkpure.com.apk 
I: Using Apktool 2.4.2-f545c2-SNAPSHOT on Vivy_v1.52.2_apkpure.com.apk
I: Loading resource table...
I: Decoding AndroidManifest.xml with resources...
I: Loading resource table from file: /home/ibotpeaches/.local/share/apktool/framework/1.apk
I: Regular manifest package...
I: Decoding file-resources...
I: Decoding values */* XMLs...
I: Baksmaling classes.dex...
I: Baksmaling classes2.dex...
I: Baksmaling classes3.dex...
I: Baksmaling classes4.dex...
I: Baksmaling classes5.dex...
I: Baksmaling classes6.dex...
I: Baksmaling classes7.dex...
I: Baksmaling classes8.dex...
I: Baksmaling classes9.dex...
I: Copying assets and libs...
I: Copying unknown files...
I: Copying original files...
I: Copying META-INF/services directory
➜  1991
➜  1991 apktool b Vivy_v1.52.2_apkpure.com --use-aapt2
I: Using Apktool 2.4.2-f545c2-SNAPSHOT
I: Checking whether sources has changed...
I: Checking whether sources has changed...
I: Checking whether sources has changed...
I: Checking whether sources has changed...
I: Checking whether sources has changed...
I: Checking whether sources has changed...
I: Checking whether sources has changed...
I: Checking whether sources has changed...
I: Checking whether sources has changed...
I: Checking whether resources has changed...
I: Building resources...
I: Copying libs... (/lib)
I: Copying libs... (/kotlin)
I: Copying libs... (/META-INF/services)
I: Building apk file...
I: Copying unknown files/dir...
I: Built apk...
➜  1991 
```

Sorry for delay. I can't replicate with latest (soon to be released) v2.5.0.

So closing.

**Comment by Mrrk23** [comment]
@ta1bbty  Bro I had the same error just exactly like yours and you wouldn't believe me. I just closed that terminal and open again and then run.. It works. I am having apktool version 2.5.0-dirty.
